contracts/src/BoundlessMarket.sol

```diff
@@ -432,11 +432,7 @@ contract BoundlessMarket is
         }
 
         if (paymentError.length > 0) {
-            if (fill.requirePayment) {
-                revertWith(paymentError);
-            } else {
-                emit PaymentRequirementsFailed(paymentError);
-            }
+            revertWith(paymentError);
```

Comment:
Sorry, this was not explained well in the issue. We actually want to always apply `requirePayment = false` rather than `requirePayment = true` 🙈
```suggestion
            emit PaymentRequirementsFailed(paymentError);
```

Reply:
When you simulate a call you only have read permissions so any event would not be available. As such I dropped the event and instead returned an error

Reply:
My bad (I wrote the issue) Sorry this wasn't clear!